content/Animations in React Native - Fernanda Toledo.md

```diff
@@ -0,0 +1,114 @@
+ 
+<script src="https://gist.github.com/fernandatoledo/3a656f21eb16f01b0f963a7fec1fb09e.js"></script>
+
+
+What we are doing here is creating a variable that will contain the value that we will be animating. We use `useRef` since the returned value will persist for the full lifetime of the component.
+Then we take that variable and feed it to the `animated.timing()` method that transitions the initial value we set in the variable to the final value we provide in `toValue`.
+By passing the current value of the ref variable to a style property of a given animated component, said component is re-rendered on a high priority thread, which will make your animations run smoothly.
+Did you get lost already? The same happened to us, that's why we created our hook library.
+
+### Our first animation hook
+
+Given our experience, we've identified a repetitive pattern among React Native animations. Opacity, translations, rotations, size, scale, and background color transition. All of these animations share very similar code. That's why we started this project by creating a simple hook that is much easier to use in hopes that more developers would be willing to give animations a try. The hook covers the basics but you would be surprised how much value a simple animation can add.
+
+Let's take a look at the first example we provided above but this time using the `useAnimate` hook:
+<script src="https://gist.github.com/aguscha333/956eeed3d174351f8fddde25aa60888a.js"></script>
+
+
+As you can see with just a few simple lines you can have your opacity animation. With the same simplicity, you can start animating background colors, translations to get objects moving, rotations, and much more.
+
+#### How to handle a bit more complex animations?
+
+Although the useAnimate hook covers most transitions that you would want to do, there are some limitations to it. The useAnimate hook is most useful for atomic transitions and by that we mean that it is perfect for animating one thing at a time. If you want to run parallel animations or sequences, you might need to use some functionalities that the basic hook simply does not cover. For these cases, we have two extra hooks that can help.
+
+This project started as just one basic animation hook. Later on, while developing a demo app for showcasing purposes, we started fiddling around with what the Animated API has to offer and we ended up creating two more hooks, one for running parallel animations and one for running them in sequence.
+
+Here are examples of how using those hooks might look like versus their plain Animated API counterparts. 
+
+##### Example of sequence animation
+
+![](https://media.giphy.com/media/hQcYhmrFy7aXFReeR7/giphy.gif)
+
+<script src="https://gist.github.com/fernandatoledo/4e0b35f95745bdc0ea3ab8ba24cc1776.js"></script>
+
+<script src="https://gist.github.com/fernandatoledo/2b2813757178ecae28851457e15b1160.js"></script>
```

Comment:
maybe you can set a subtitle saying this is with our hook and this is without our hook. Also, since before you first show the no hook way, I would change the order here to be consistent (right now the first gist is using your hook and the second one without the hook)

Reply:
Also, in the blog channel said you should just put the gists URL, without any <script> tag.

Reply:
Awesome! Thanks :)